server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -281,18 +277,16 @@ public InternalEngine(EngineConfig engineConfig) {
                 );
                 this.localCheckpointTracker = createLocalCheckpointTracker(localCheckpointTrackerSupplier);
                 // TODO: Segrep - should have a separate read only engine rather than all this conditional logic.
-                if (isReadOnlyReplica()) {
+                if (engineConfig.isReadOnly()) {
                     // Segrep - hack to make this engine read only and not use
                     writer = null;
-                    historyUUID = null;
-                    forceMergeUUID = null;
                 } else {
                     writer = createWriter();
                     bootstrapAppendOnlyInfoFromWriter(writer);
-                    final Map<String, String> commitData = commitDataAsMap(writer);
-                    historyUUID = loadHistoryUUID(commitData);
-                    forceMergeUUID = commitData.get(FORCE_MERGE_UUID_KEY);
                 }
+                final Map<String, String> commitData = commitDataAsMap(writer);
+                historyUUID = loadHistoryUUID(commitData);
+                forceMergeUUID = commitData.get(FORCE_MERGE_UUID_KEY);
```

Comment:
What's the difference here? Is it just a style check?

Reply:
we don't need to conditionally set these anymore, they will correctly be set in both situations.  histouryUUID will be set in the user data returned with the latest commit.  forceMergeUUID will not be in the map here so it will be set to null.